node/network/protocol/src/peer_set.rs

```diff
@@ -0,0 +1,115 @@
+///
+/// Ideally this would be defined in substrate as a newtype.
+type ProtocolName = Cow<'static, str>;
+
+impl PeerSet {
+	/// Get `sc_network` peer set configurations for each peerset.
+	///
+	/// Those should be used in the network configuration to register the protocols with the
+	/// network service.
+	pub fn get_info(self) -> NonDefaultSetConfig {
+		let protocol = self.into_protocol_name();
+		match self {
+			PeerSet::Validation => NonDefaultSetConfig {
+				notifications_protocol: protocol,
+				set_config: sc_network::config::SetConfig {
+					in_peers: 25,
+					out_peers: 0,
+					reserved_nodes: Vec::new(),
+					non_reserved_mode: sc_network::config::NonReservedPeerMode::Accept,
+				},
+			},
+			PeerSet::Collation => NonDefaultSetConfig {
+				notifications_protocol: protocol,
+				set_config: SetConfig {
+					in_peers: 25,
+					out_peers: 0,
+					reserved_nodes: Vec::new(),
+					non_reserved_mode: sc_network::config::NonReservedPeerMode::Accept,
+				},
+			},
+            PeerSet::AvailabilityDistribution => NonDefaultSetConfig {
+                notifications_protocol: protocol,
+                set_config: sc_network::config::SetConfig {
+                    in_peers: 25,
+                    out_peers: 0,
+                    reserved_nodes: Vec::new(),
+                    non_reserved_mode: sc_network::config::NonReservedPeerMode::Accept,
+                },
+            },
+		}
+	}
+
+	/// Get the protocol name associated with each peer set as static str.
+	pub const fn get_protocol_name_static(self) -> &'static str {
+		match self {
+			PeerSet::Validation => "/polkadot/validation/1",
+			PeerSet::Collation => "/polkadot/collation/1",
+            PeerSet::AvailabilityDistribution => "/polkadot/availability/1",
+		}
+	}
+
+	/// Convert a peer set into a protocol name as understood by Substrate.
+	///
+	/// With `ProtocolName` being a proper newtype we could use the `Into` trait here.
+	pub fn into_protocol_name(self) -> ProtocolName {
+		self.get_protocol_name_static().into()
+	}
+
+	/// Try parsing a protocol name into a peer set.
+	///
+	/// If ProtocolName was a newtype, this would actually be nice to implement in terms of the
+	/// standard `TryFrom` trait.
+	pub fn try_from_protocol_name(name: &ProtocolName) -> Option<PeerSet> {
+		match name {
+			n if n == &PeerSet::Validation.into_protocol_name() => Some(PeerSet::Validation),
+			n if n == &PeerSet::Collation.into_protocol_name() => Some(PeerSet::Collation),
+            n if n == &PeerSet::AvailabilityDistribution.into_protocol_name() => {
+                Some(PeerSet::AvailabilityDistribution)
+            }
+			_ => None,
+		}
+	}
+}
+
+/// Get `NonDefaultSetConfig`s for all available peer sets.
+///
+/// Should be used during network configuration (added to [`NetworkConfiguration::extra_sets`])
+/// or shortly after startup to register the protocols with the network service.
+pub fn peer_set_infos() -> Vec<sc_network::config::NonDefaultSetConfig> {
```

Comment:
```suggestion
pub fn peer_set_info() -> Vec<sc_network::config::NonDefaultSetConfig> {
```

Nit: "information" is inherently plural; "infos" sounds weird.

Reply:
Fair :-) Ok I will go with peer_sets_info then.